src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithms.cs

```diff
@@ -78,13 +78,105 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
             analysisContext.RegisterCompilationStartAction(
-                context =>
+                (CompilationStartAnalysisContext context) =>
                 {
                     var cryptTypes = new CompilationSecurityTypes(context.Compilation);
-                    if (ReferencesAnyTargetType(cryptTypes))
+                    if (!ReferencesAnyTargetType(cryptTypes))
                     {
-                        GetAnalyzer(context, cryptTypes);
+                        return;
                     }
+
+                    context.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            IMethodSymbol method;
+
+                            switch (operationAnalysisContext.Operation)
+                            {
+                                case IInvocationOperation invocationOperation:
+                                    method = invocationOperation.TargetMethod;
+                                    break;
+                                case IObjectCreationOperation objectCreationOperation:
+                                    method = objectCreationOperation.Constructor;
+                                    break;
+                                default:
+                                    Debug.Fail($"Unhandled IOperation {operationAnalysisContext.Operation.Kind}");
+                                    return;
+                            }
+
+                            INamedTypeSymbol type = method.ContainingType;
+                            string[] messageArgs = new string[2];
+                            DiagnosticDescriptor rule = null;
+
+                            messageArgs[0] = operationAnalysisContext.ContainingSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
```

Comment:
We are trying to avoid full type names in diagnostic messages - these end up getting way too long for very common code bases, making the diagnostic text very hard to read. Can we just use `ContainingSymbol.Name` and rely on the diagnostic location/squiggle to do the rest?

Reply:
Sure.